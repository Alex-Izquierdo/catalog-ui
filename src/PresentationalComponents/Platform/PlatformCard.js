import React from 'react';
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { GridItem, Card, CardHeader, CardBody, CardFooter } from '@patternfly/react-core';
import DefaultPlatformImg from '../../assets/images/platform-default.svg';
import OpenshiftPlatformImg from '../../assets/images/platform-openshift.svg';
import AmazonPlatformImg from '../../assets/images/platform-amazon.png';
import ImageWithDefault from '../Shared/ImageWithDefault';
import ItemDetails from '../Shared/CardCommon';
import './platformcard.scss';

const TO_DISPLAY = [ 'description', 'modified' ];

// TO DO - use webpack to load all images
const platformTypeImg = (sourceTypeId) => {
  switch (sourceTypeId) {
    case '1':
      return OpenshiftPlatformImg;
    case '2':
      return AmazonPlatformImg;
    default:
      return DefaultPlatformImg;
  }
};

const PlatformCard = ({ name, id, ...props }) => (
  <GridItem sm={ 6 } md={ 4 } lg={ 4 } xl={ 3 }>
    <Link to={ `/platform/${id}` } className="card-link">
      <Card key={ id }>
        <CardHeader className="pcard_header">
          <ImageWithDefault src={ platformTypeImg(props.source_type_id || '0') } defaultSrc={ DefaultPlatformImg } width="80" height="40"/>
        </CardHeader>
        <CardBody className="pcard_body">
          <h4>{ name }</h4>
          <ItemDetails { ...{ name, ...props } } toDisplay={ TO_DISPLAY } />
        </CardBody>
        <CardFooter/>
      </Card>
    </Link>
  </GridItem>
);

PlatformCard.propTypes = {
  history: propTypes.object,
  imageUrl: propTypes.string,
  name: propTypes.string,
  source_type_id: propTypes.string,
  id: propTypes.string
};

export default PlatformCard;
