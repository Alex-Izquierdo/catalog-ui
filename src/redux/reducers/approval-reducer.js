import { SET_LOADING_STATE } from '../action-types';
import { ASYNC_ACTIONS } from '../action-types/approval-action-types';

export const approvalInitialState = {
  isFetching: false,
  isResolving: false,
  workflows: [],
  resolvedWorkflows: []
};

const setLoadingState = (state, { payload = true }) => ({
  ...state,
  isFetching: payload
});
const setWorkflows = (state, { payload }) => ({
  ...state,
  isFetching: false,
  workflows: payload
});
const setResolvingState = (state, { payload = true }) => ({
  ...state,
  isResolving: payload
});
const setResolvedWorkflows = (state, { payload }) => ({
  ...state,
  isResolving: false,
  resolvedWorkflows: payload
});

export default {
  [ASYNC_ACTIONS.FETCH_WORKFLOWS_PENDING]: setLoadingState,
  [ASYNC_ACTIONS.FETCH_WORKFLOWS_FULFILLED]: setWorkflows,
  [ASYNC_ACTIONS.RESOLVE_WORKFLOWS_PENDING]: setResolvingState,
  [ASYNC_ACTIONS.RESOLVE_WORKFLOWS_FULFILLED]: setResolvedWorkflows,
  [SET_LOADING_STATE]: setLoadingState
};
