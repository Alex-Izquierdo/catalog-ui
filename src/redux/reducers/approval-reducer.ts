import { Workflow } from '@redhat-cloud-services/approval-client';
import { SET_LOADING_STATE } from '../action-types';
import { ASYNC_ACTIONS } from '../action-types/approval-action-types';
import { StateFromReducersMapObject } from 'redux';
import { ReduxAction, AnyObject } from '../../types/common-types';

export interface ApprovalReducerState extends AnyObject {
  isFetching: boolean;
  isResolving: boolean;
  workflows: Workflow[];
  resolvedWorkflows: Workflow[];
}

export type ApprovalReducerActionHandler = (
  state: StateFromReducersMapObject<ApprovalReducerState>,
  action: ReduxAction
) => ApprovalReducerState;

export const approvalInitialState = {
  isFetching: false,
  isResolving: false,
  workflows: [],
  resolvedWorkflows: []
};

const setLoadingState: ApprovalReducerActionHandler = (
  state,
  { payload = true }
) => ({
  ...state,
  isFetching: payload
});
const setWorkflows: ApprovalReducerActionHandler = (state, { payload }) => ({
  ...state,
  isFetching: false,
  workflows: payload
});
const setResolvingState: ApprovalReducerActionHandler = (
  state,
  { payload = true }
) => ({
  ...state,
  isResolving: payload
});
const setResolvedWorkflows: ApprovalReducerActionHandler = (
  state,
  { payload }
) => ({
  ...state,
  isResolving: false,
  resolvedWorkflows: payload
});

export default {
  [ASYNC_ACTIONS.FETCH_WORKFLOWS_PENDING]: setLoadingState,
  [ASYNC_ACTIONS.FETCH_WORKFLOWS_FULFILLED]: setWorkflows,
  [ASYNC_ACTIONS.RESOLVE_WORKFLOWS_PENDING]: setResolvingState,
  [ASYNC_ACTIONS.RESOLVE_WORKFLOWS_FULFILLED]: setResolvedWorkflows,
  [SET_LOADING_STATE]: setLoadingState
};
