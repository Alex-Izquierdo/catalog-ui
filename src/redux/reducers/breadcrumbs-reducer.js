import { INITIALIZE_BREADCRUMBS } from '../action-types';

export const initialBreadcrumbsState = {
  fragments: []
};

const initialize = (state, { payload }) => ({ ...state, fragments: payload });

export default {
  [INITIALIZE_BREADCRUMBS]: initialize
};
