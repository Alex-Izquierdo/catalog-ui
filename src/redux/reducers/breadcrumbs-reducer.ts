import { ReactNode } from 'react';
import { StateFromReducersMapObject } from 'redux';
import { INITIALIZE_BREADCRUMBS } from '../action-types';
import { AnyObject, ReduxAction } from '../../types/common-types';

export interface BreadcrumbFragment {
  pathname: string;
  searchParams?: string;
  title?: ReactNode;
}
export interface BreadcrumbsReducerState extends AnyObject {
  fragments: BreadcrumbFragment[];
}

export type BreadcrumbsReducerActionHandler = (
  state: StateFromReducersMapObject<BreadcrumbsReducerState>,
  action: ReduxAction
) => BreadcrumbsReducerState;

export const initialBreadcrumbsState: BreadcrumbsReducerState = {
  fragments: []
};

const initialize: BreadcrumbsReducerActionHandler = (state, { payload }) => ({
  ...state,
  fragments: payload
});

export default {
  [INITIALIZE_BREADCRUMBS]: initialize
};
