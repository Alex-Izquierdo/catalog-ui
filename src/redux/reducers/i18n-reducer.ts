import { IntlShape } from 'react-intl';
import { INITIALIZE_I18N } from '../action-types';
import { AnyObject, ReduxActionHandler } from '../../types/common-types';

export type I18nReducerState = Partial<IntlShape> & AnyObject;
export type I18nReducerActionHandler = ReduxActionHandler<I18nReducerState>;

export const i18nInitialState: I18nReducerState = {
  formatMessage: ({ defaultMessage = '' }) => defaultMessage
};
const initialize: I18nReducerActionHandler = (state, { payload }) => ({
  ...state,
  ...payload
});

export default {
  [INITIALIZE_I18N]: initialize
};
