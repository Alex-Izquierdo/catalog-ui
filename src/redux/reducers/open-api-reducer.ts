import { SET_OPENAPI_SCHEMA } from '../action-types';
import { AnyObject, ReduxActionHandler } from '../../types/common-types';

export interface OpenApiReducerState extends AnyObject {
  schema?: AnyObject;
}
export type OpenApiReducerActionHandler = ReduxActionHandler<
  OpenApiReducerState
>;

export const openApiInitialState: OpenApiReducerState = {
  schema: undefined
};
const setSchema: OpenApiReducerActionHandler = (state, { payload }) => ({
  ...state,
  schema: payload
});

export default {
  [SET_OPENAPI_SCHEMA]: setSchema
};
