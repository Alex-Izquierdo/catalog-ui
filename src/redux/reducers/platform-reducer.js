import {
  FETCH_PLATFORM,
  FETCH_PLATFORMS,
  FETCH_PLATFORM_ITEMS,
  FETCH_PLATFORM_ITEM,
  FILTER_PLATFORM_ITEMS,
  FETCH_MULTIPLE_PLATFORM_ITEMS,
  SET_LOADING_STATE,
  FETCH_PLATFORM_INVENTORIES,
  SET_INVENTORIES_LOADING_STATE
} from '../action-types';

// Initial State
export const platformInitialState = {
  isPlatformDataLoading: false,
  selectedPlatform: {},
  platforms: [],
  platformItems: {},
  platformInventories: [],
  platformItem: {},
  platform: {},
  filterValue: ''
};

// rename isPlatformLoading.. to isLoading so we can use common action for loading states

const setLoadingState = (state, { payload = true }) => ({
  ...state,
  isPlatformDataLoading: payload
});
const setPlatforms = (state, { payload }) => ({
  ...state,
  platforms: payload,
  isPlatformDataLoading: false
});
const setPlatformItems = (state, { payload, meta: { platformId } }) => ({
  ...state,
  platformItems: { ...state.platformItems, [platformId]: payload },
  isPlatformDataLoading: false
});
const setMultiplePlatformItems = (state, { payload }) => ({
  ...state,
  platformItems: { ...state.platformItems, ...payload },
  isPlatformDataLoading: false
});
const setPortfolioItems = (state, { payload }) => ({
  ...state,
  portfolioItem: payload,
  isPlatformDataLoading: false
});
const selectPlatform = (state, { payload }) => ({
  ...state,
  selectedPlatform: payload,
  isLoading: false
});
const filterPlatformItems = (state, { payload }) => ({
  ...state,
  filterValue: payload
});
const setInventoriesDataLoadingState = (state, { payload = true }) => ({
  ...state,
  isInventoriesDataLoading: payload
});
const setPlatformInventories = (state, { payload }) => ({
  ...state,
  platformInventories: { ...state.platformInventories, ...payload },
  isInventoriesDataLoading: false
});

export default {
  [`${FETCH_PLATFORMS}_PENDING`]: setLoadingState,
  [`${FETCH_PLATFORMS}_FULFILLED`]: setPlatforms,
  [`${FETCH_PLATFORM_ITEMS}_PENDING`]: setLoadingState,
  [`${FETCH_PLATFORM_ITEMS}_FULFILLED`]: setPlatformItems,
  [`${FETCH_PLATFORM_ITEM}_PENDING`]: setLoadingState,
  [`${FETCH_PLATFORM_ITEM}_FULFILLED`]: setPortfolioItems,
  [`${FETCH_PLATFORM}_PENDING`]: setLoadingState,
  [`${FETCH_PLATFORM}_FULFILLED`]: selectPlatform,
  [`${FILTER_PLATFORM_ITEMS}_FULFILLED`]: filterPlatformItems,
  [`${FETCH_MULTIPLE_PLATFORM_ITEMS}_FULFILLED`]: setMultiplePlatformItems,
  [SET_LOADING_STATE]: setLoadingState,
  [`${FETCH_PLATFORM_INVENTORIES}_PENDING`]: setInventoriesDataLoadingState,
  [`${FETCH_PLATFORM_INVENTORIES}_FULFILLED`]: setPlatformInventories,
  [SET_INVENTORIES_LOADING_STATE]: setInventoriesDataLoadingState
};
