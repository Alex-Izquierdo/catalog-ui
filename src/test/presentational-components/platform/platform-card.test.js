import React from 'react';
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import promiseMiddleware from 'redux-promise-middleware';
import notificationsMiddleware from '@redhat-cloud-services/frontend-components-notifications/notificationsMiddleware';
import { platformInitialState } from '../../../redux/reducers/platform-reducer';

import PlatformCard from '../../../presentational-components/platform/platform-card';

describe('<PlatformCard />', () => {
  let initialProps;
  let initialState;
  const middlewares = [thunk, promiseMiddleware, notificationsMiddleware()];
  let mockStore;

  const ComponentWrapper = ({ store, children }) => (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  );

  beforeEach(() => {
    initialProps = {
      id: 'Foo'
    };
    mockStore = configureStore(middlewares);
    initialState = {
      platformReducer: { ...platformInitialState, isLoading: false }
    };
  });

  it('should render correctly', (done) => {
    const store = mockStore(initialState);
    const wrapper = mount(
      <ComponentWrapper store={store}>
        <PlatformCard {...initialProps} />
      </ComponentWrapper>
    );
    expect(shallowToJson(wrapper.find(PlatformCard))).toMatchSnapshot();
    done();
  });
});
