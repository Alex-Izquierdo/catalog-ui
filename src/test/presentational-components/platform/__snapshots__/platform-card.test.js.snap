// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlatformCard /> should choose card image 1`] = `
<PlatformCard
  description="desc"
  modified="Foo"
>
  <GalleryItem>
    <div>
      <Link
        className="card-link"
        replace={false}
        to="/platforms/detail/undefined"
      >
        <a
          className="card-link"
          href="/platforms/detail/undefined"
          onClick={[Function]}
        >
          <Card
            className="content-gallery-card"
            component="article"
          >
            <article
              className="pf-c-card content-gallery-card"
            >
              <CardHeader
                className=""
                component="div"
              >
                <div
                  className="pf-c-card__header"
                >
                  <ImageWithDefault
                    height="40"
                    src="test-file-stub"
                    width="80"
                  >
                    <img
                      height="40"
                      src="test-file-stub"
                      width="80"
                    />
                  </ImageWithDefault>
                </div>
              </CardHeader>
              <CardBody
                className=""
                component="div"
              >
                <div
                  className="pf-c-card__body"
                >
                  <h4 />
                  <ItemDetails
                    description="desc"
                    modified="Foo"
                    toDisplay={
                      Array [
                        "description",
                        "modified",
                      ]
                    }
                  >
                    <div
                      className="line-clamp"
                    >
                      <PropLine
                        key="card-prop-description"
                        value="desc"
                      >
                        <div>
                          desc
                        </div>
                      </PropLine>
                      <PropLine
                        key="card-prop-modified"
                        value="Foo"
                      >
                        <div>
                          Foo
                        </div>
                      </PropLine>
                    </div>
                  </ItemDetails>
                </div>
              </CardBody>
              <CardFooter
                className=""
                component="div"
              >
                <div
                  className="pf-c-card__footer"
                />
              </CardFooter>
            </article>
          </Card>
        </a>
      </Link>
    </div>
  </GalleryItem>
</PlatformCard>
`;

exports[`<PlatformCard /> should render correctly 1`] = `
<PlatformCard
  description="desc"
  modified="Foo"
>
  <GalleryItem>
    <div>
      <Link
        className="card-link"
        replace={false}
        to="/platforms/detail/undefined"
      >
        <a
          className="card-link"
          href="/platforms/detail/undefined"
          onClick={[Function]}
        >
          <Card
            className="content-gallery-card"
            component="article"
          >
            <article
              className="pf-c-card content-gallery-card"
            >
              <CardHeader
                className=""
                component="div"
              >
                <div
                  className="pf-c-card__header"
                >
                  <ImageWithDefault
                    height="40"
                    src="test-file-stub"
                    width="80"
                  >
                    <img
                      height="40"
                      src="test-file-stub"
                      width="80"
                    />
                  </ImageWithDefault>
                </div>
              </CardHeader>
              <CardBody
                className=""
                component="div"
              >
                <div
                  className="pf-c-card__body"
                >
                  <h4 />
                  <ItemDetails
                    description="desc"
                    modified="Foo"
                    toDisplay={
                      Array [
                        "description",
                        "modified",
                      ]
                    }
                  >
                    <div
                      className="line-clamp"
                    >
                      <PropLine
                        key="card-prop-description"
                        value="desc"
                      >
                        <div>
                          desc
                        </div>
                      </PropLine>
                      <PropLine
                        key="card-prop-modified"
                        value="Foo"
                      >
                        <div>
                          Foo
                        </div>
                      </PropLine>
                    </div>
                  </ItemDetails>
                </div>
              </CardBody>
              <CardFooter
                className=""
                component="div"
              >
                <div
                  className="pf-c-card__footer"
                />
              </CardFooter>
            </article>
          </Card>
        </a>
      </Link>
    </div>
  </GalleryItem>
</PlatformCard>
`;
